geode-core/src/main/java/org/apache/geode/cache/client/internal/PoolImpl.java

```diff
@@ -361,6 +358,7 @@ public CancelCriterion getCancelCriterion() {
   }
 
   @Override
+  @Deprecated
```

Comment:
Does this matter if we've already deprecated the public API?

Reply:
Kinda, if the interfaces is deprecated but the implementation is not then internally where we use the implementation the compiler and IJ don’t warn about the use. Adding to here too makes warnings propagate to the implementation as well. Silly.

Reply:
Solved this with @dschneider-pivotal suggestion to use default.